src/main/java/com/aws/greengrass/logmanager/model/LogFile.java

```diff
@@ -85,16 +78,19 @@ public String hashString() {
             if (!this.exists()) {
                 return fileHash;
             }
-            List<String> lines = getLines();
-            if (lines.size() < linesNeeded) {
-                return fileHash;
+            String bytesReadString = new String(readBytes(), StandardCharsets.UTF_8);
+            // if the we read less then 1KB but it contains an entire line, it is still valid.
+            if (didRead < bytesNeeded && bytesReadString.indexOf('\n') > -1) {
```

Comment:
Use variable names that represent your intent

```suggestion
    boolean hasLessThanBytesNeeded = didRead < bytesNeeded
    boolean containsEntireLine = bytesReadString.indexOf('\n') > -1
    if (hasLessThanBytesNeeded && containsEntireLine) {
```

Reply:
What if the log file just has an empty line?, do we really care about it being a full line? - I think what we should care about is we have read "enough bytes" so we should read hopefully all the ones we require but if there are less than that guarantee that they are more than a minimum

Reply:
I think there are two reasons:
1. The case of empty line is rare, because it is not a standard way to write logs.  
2. There may be the case that the log file only has content < 1KB, but we should not miss uploading these contents.

Reply:
1) We should handle it and not make that assume it wont happen.

2) Why wouldn't we want to create a hash if we have less bytes than what we want but not a full line? - I think it is better to require at least a minimum instead of expecting a new line

Reply:
It is hard to define the minimum number. For example, the log file may only has one line, and it only contains 30 bytes. Then the file is rotated. In this example, we still need to upload this line since it is actually an valid log line. But it is hard for us to say 30 is the right number as the minimum number.

Reply:
I think we are mixing 2 separate things here:

1) The how many bytes we need to compute a hash has nothing to do with the fact that the file might have a single line or multiple lines. To address your example we might have a file with a single line and not line break that is below the bytes you require that gets rotated and we still would have to upload it.

2) The other problem with this is that calculating the hash with this implementation is not deterministic - Depending on when you call `hashString` it might return different values
     2.1) If you call it and you have less than what you require it will return a value
     2.2) If the file then get more lines written to it and you can get all the bytes you need you will get a different hash.


We either have to memoize the hash value after you first compute it or have a firm required number of bytes you need.

Reply:
Agree. I will change it to the following:
1. If there is a new line before 1KB, then we take a new line
2. Otherwise, we read 1KB to prevent crazy long one line message.